src/ErrorMessage.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Validator;
+
+use Yiisoft\Translator\TranslatorInterface;
+
+final class ErrorMessage
+{
+    public string $message = '';
+    public array $params = [];
+
+    private ?TranslatorInterface $translator;
+
+    public function __construct(string $message, ?array $params = null, ?TranslatorInterface $translator = null)
+    {
+        $this->message = $message;
+        $this->params = $params?? [];
+        $this->translator = $translator;
+    }
+
+    public function withTranslator(?TranslatorInterface $translator = null): self
+    {
+        if ($translator !== null) {
```

Comment:
There's no need for this condition. It prevents removing of translator.

Reply:
I don't understand. If remove this line - $this->translator not be used with empty param $translator

Reply:
1. The method accepts `null`. I'd suppose it means "use no translator".
2. If I'll do `(new ErrorMessage('bla', [], $translator))->withTranslator(null)` with current code it won't remove translator. It will stil return same instance with `$translator` assigned.